### ISSUE TITLE
Restrict Uniques for Officers in relation to Phoenix Home Title

### ISSUE THREAD

**Issue by BrianisBlack** [BSData/star-wars-armada#21]
Phoenix Home title allows users to select two officers in separate slots.  Much like the modification restriction, a condition filter needs added to remove uniques from the officer slot if they're selected in the first one.

**Comment by KealL** [comment]
I can do this Wednesday unless you want to do this before then :)

**Comment by XWingDrLucky** [comment]
This doesn't make any sense to validate the crew the same way mods are validated.  All upgrades are restricted to one per ship (See FAQ page 2).  The modifications are screwy because they appear in different slots.  This is much simpler - all officers should be set to "Max 1 Selections in Parent" and then when the "Phoenix Home" title is selected, have it change the max selections  of officers to 2.

Frankly, all upgrades for this game should be set to "Max 1 Selections in Parent" because a ship can never equip more than one copy of the same upgrade card.  That would also mean things like giving the MC80 Assault a single Defensive Retrofit Entry Group with two selections allowed rather than two separate Entry Groups (same with all other ships that have more than one copy of a single upgrade slot).  That would do a lot to untangle upgrade validation.

**Comment by KealL** [comment]
I agree and disagree at the same time Br Lucky
As a user I feel 2 selection entries is much better than increasing the max selections to 2, as its a lot clearer that you can pick TWO from that group with the separate slots.
However, I agree a simpler method of managing the data would be good.

So I propose:
-We add a new cost base call Mod, and then restrict each ship to a max of 1 mod and set each mod to cost 1
-We add the max 1 selections in parent to every upgrade card.

This will then raise a flag if two mods are selected and enable people to tell which are the mods as it will cost x pts and 1 Mod

Thougs?

**Comment by BrianisBlack** [comment]
I'm not aware of an easy way to introduce separate variables to track, I recall trying to do something similar to that for Mods initially, but wasn't able to find anything that was as simple to implement as what @KealL did.  

On increasing max selections vs. multiple entries, I feel like multiple entries are more user-friendly, but it might also be good to use the same logic as the x-wing files to keep the behavior the same between the two systems.

**Comment by XWingDrLucky** [comment]
To be clear, I'm not suggesting changing how Mods are tracked. It's
irregular how they are restricted and I think the current method works for
them (though it could get unwieldy if tons more mods were released). I'm
only talking about how to validate the "only one of any given upgrade per
ship" rule on ships that can take more than one of a particular type (as
this was raised because of the Phoenix Home). Note that in general, users
will be able to tell easily if they are allowed more than one selection
because the radio buttons will change to check boxes.

Also, I know I haven't been to active on this side of the house (mostly
sticking to the X-Wing side), but I finally picked up Armada, so I may be
around more. Let me know how I can help.

**Comment by XWingDrLucky** [comment]
I went ahead and tagged every upgrade in the catalogues with 1 per parent and grouped together the entry groups that had multiple versions.  In doing so, it started breaking the Modification validation, which I knew would happen and planned to fix.  However, in starting to repair the Modification validation, I came up with a much cleaner way to do it.

I built a single shared entry in the gst called "Modification" with min 1/max 1 in parent (so it automatically selects when attached to something).  I linked the "Modification" shared entry to every Modification Upgrade card and told each of them to set hidden to true if Parent had at least one instance of "Modification" and zero instances of the given upgrade like such (Using Expanded Launchers as an example):

Expanded Launchers                                                                                 <---- Selection Entry
- max 1 Selections in Parent                                                                     <---- Constraint
- Link to: Modification (Entry)                                                                    <---- Entry Link
- Expanded Launchers                                                                               <---- Profile
- set Hidden to true                                                                                   <---- Modifier
- - (and)                                                                                                      <---- Condition Group
- - - If Equal to 0 selections in Parent of Expanded Launchers (Shared)    <---- Condition
- - - If At Least 1 selections in Parent of Modification (Shared)                 <---- Condition

By building each Modification like this, when one is selected, all other Modifications available to that ship will hide, but we avoid having to link each one.  As an added bonus, in the Roster Editor, the added upgrade shows it has something attached to it called "Modification" which should clue in the user why things are being constrained.  It seems to be working well.  I didn't set out to change Modifications but this seems elegant so I went with it.
